pkg/klusterlet/clusterinfo/clusterinfo_controller.go

```diff
@@ -388,6 +399,52 @@ func (r *ClusterInfoReconciler) getDistributionInfo() (clusterv1beta1.Distributi
 	return distributionInfo, nil
 }
 
+func distributionInfoUpdated(old, new *clusterv1beta1.DistributionInfo) bool {
+	switch new.Type {
+	case clusterv1beta1.DistributionTypeOCP:
+		return ocpDistributionInfoUpdated(&old.OCP, &new.OCP)
+	}
+	return false
+}
+
+func ocpDistributionInfoUpdated(old, new *clusterv1beta1.OCPDistributionInfo) bool {
+	if len(old.AvailableUpdates) != len(new.AvailableUpdates) {
+		return true
+	}
+	if old.AvailableUpdates != nil && new.AvailableUpdates != nil {
+		sort.SliceStable(old.AvailableUpdates, func(i, j int) bool { return old.AvailableUpdates[i] < old.AvailableUpdates[j] })
```

Comment:
Why do we need to sort the old one? I think it  is sorted already in the last update

Reply:
my understanding is that we can't guarantee that the old one is sorted  after marshal/Unmarshal. 
use nodeStatus as an example, the original issue happened when nodeStatus in clusterinfo are different from the ones got from nodeList (slice order is different) although they have been updated using the last ones got from nodeList.

Reply:
but the old one is fetched from managedclusterinfo which should be sorted before write,  right?

Reply:
verified that the clusterinfo is not out of order. deleted the part to sort old one.